infrastructure/generators/util.py

```diff
@@ -139,3 +140,100 @@ def inner():
         return f"{article1} {subject} {predicate} {article2} {direct_object}."
 
     return inner
+
+
+op_parts = [
+    "celery",
+    "task",
+    "redis",
+    "feature",
+    "flagr",
+    "has",
+    "nodestore",
+    "set_subkeys",
+    "sentry",
+    "reprocessing2",
+    "save_unprocessed_event",
+    "get_unprocessed_event",
+    "relay",
+    "actors",
+    "connector",
+    "controller",
+    "events",
+    "healthcheck",
+    "mod",
+    "outcome",
+    "project",
+    "project_cache",
+    "project_local",
+    "project_redis",
+    "project_upstream",
+    "relay",
+    "server",
+    "store",
+    "upstream",
+]
+
+
+def op_generator():
+    def inner():
+        num_segments = random.randrange(1, 5)
+        segments = []
+        while len(segments) < num_segments:
+            segment = random.choice(op_parts)
+            if segment not in segments:
+                segments.append(segment)
+        return ".".join(segments)
+
+    return inner
+
+
+def trace_generator(
+    trace_release=None,
+    trace_user_id=None,
+    trace_user_segment=None,
+    trace_environment=None,
+    public_key=None,
+    **kwargs
+):
+    """
+    Generates the trace header to be placed in an envelope header
+
+    """
+    if trace_user_id is not None or trace_user_segment is not None:
+        user_generator = {}
+        if trace_user_segment is not None:
+            user_generator["segment"] = trace_user_segment
+        if trace_user_id is not None:
+            user_generator["id"] = trace_user_id
+    else:
+        user_generator = None
+    return schema_generator(
+        release=trace_release,
+        user=user_generator,
+        environment=trace_environment,
+        trace_id=lambda: uuid.uuid4().hex,
+        public_key=public_key
+    )
+
+
+def envelope_header_generator(**kwargs):
```

Comment:
_nit_: Could make this self-documenting:

```suggestion
def envelope_header_generator(event_id=None, **kwargs):
```

Reply:
you're right (again), initially didn't have event_id param (always generating a new one) and then I thought about synchronizing it with the id from the event item and just hacked it in.